src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -207,63 +207,127 @@ public Task<Either<ActionResult, DeleteReleasePlanViewModel>> GetDeleteReleaseVe
                 });
         }
 
-        public Task<Either<ActionResult, Unit>> DeleteReleaseVersion(Guid releaseVersionId)
+        public Task<Either<ActionResult, Unit>> DeleteReleaseVersion(
+            Guid releaseVersionId,
+            CancellationToken cancellationToken = default)
         {
             return _persistenceHelper
                 .CheckEntityExists<ReleaseVersion>(releaseVersionId)
-                .OnSuccess(_userService.CheckCanDeleteReleaseVersion) // only allows unapproved amendments to be removed
+                .OnSuccess(_userService.CheckCanDeleteReleaseVersion)
                 .OnSuccessDo(async () => await _processorClient.BulkDeleteDataSetVersions(releaseVersionId))
-                .OnSuccessDo(async release => await _cacheService.DeleteCacheFolderAsync(
-                    new PrivateReleaseContentFolderCacheKey(release.Id)))
+                .OnSuccessDo(async release => await _cacheService.DeleteCacheFolderAsync(new PrivateReleaseContentFolderCacheKey(release.Id)))
                 .OnSuccessDo(async () => await _releaseDataFileService.DeleteAll(releaseVersionId))
                 .OnSuccessDo(async () => await _releaseFileService.DeleteAll(releaseVersionId))
                 .OnSuccessVoid(async releaseVersion =>
                 {
-                    // NOTE: As only removing unapproved amendments, releaseVersion.Release will not be orphaned
-                    // and associated item in releaseVersion.Publication.ReleaseSeries will also not be orphaned
-                    // so no need to remove them.
-
-                    releaseVersion.SoftDeleted = true;
-                    _context.ReleaseVersions.Update(releaseVersion);
-
-                    var roles = await _context
-                        .UserReleaseRoles
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    roles.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(roles);
-
-                    var invites = await _context
-                        .UserReleaseInvites
-                        .AsQueryable()
-                        .Where(r => r.ReleaseVersionId == releaseVersionId)
-                        .ToListAsync();
-                    invites.ForEach(r => r.SoftDeleted = true);
-                    _context.UpdateRange(invites);
-
-                    var methodologiesScheduledWithRelease =
-                        GetMethodologiesScheduledWithRelease(releaseVersionId);
-
-                    // TODO EES-2747 - this should be looked at to see how best to reuse similar "set to draft" logic
-                    // in MethodologyApprovalService.
-                    methodologiesScheduledWithRelease.ForEach(m =>
+                    if (!releaseVersion.Amendment && releaseVersion.ApprovalStatus == ReleaseApprovalStatus.Draft)
```

Comment:
Removing this causes a `ReleaseServiceTest` to fail because it doesn't (by design) go through the authorisation handler. I'd argue that the check should remain as this is re-usable business logic, i.e. even though one of the implementations includes the restriction already, there's a chance that another implementation might not.

Reply:
My thought is that if we're checking the approval status here and not relying on `CheckCanDeleteReleaseVersion`, for consistency we will also want to check it for all the preceeding `OnSuccessDo`s.

We could do something like this instead:
``` 
.OnSuccess(_userService.CheckCanDeleteReleaseVersion)
                .OnSuccess(releaseVersion =>
                {
                    if (releaseVersion.ApprovalStatus != ReleaseApprovalStatus.Draft)
                    {
                        throw new Exception("Can only delete draft releases");
                    }

                    return releaseVersion;
                })
                .OnSuccessDo(async () => await _processorClient.BulkDeleteDataSetVersions(releaseVersionId))
                [....]
```

Then the approval status is ensured across all the method calls we do, not just the final removal of the release version  itself.

Reply:
I'm still keen on removing the approval status check here - although I will accept moving it higher up. My personal opinion is that this check is redundant - for better or worse, we haven't generally kept business logic separate.

If we wanted this logic to work without the permission check so it could function as reusable business logic, I believe the logic is currently wrong. Without the permission check, it is possible to remove cache/data/general files of an approved release. And while the database entries wouldn't be removed for an approved release, they would be soft deleted.

So I'm personally inclined to remove the check entirely, but if you want this code to function as reusable business logic, then I think we need to check it's draft release first - before calling `BulkDeleteDataSetVersions`, `DeleteCacheFolderAsync`, etc. -  like the code sample I gave in the previous comment.